packages/trader/src/Modules/SmartChart/index.tsx

```diff
@@ -0,0 +1,260 @@
+type TGetContractsArray = ReturnType<typeof useStore>['contract_replay']['contract_store']['getContractsArray'];
+type TBarriers = ReturnType<typeof useStore>['portfolio']['barriers'];
+
+type TContractsArray = TMarkersArray | ReturnType<TGetContractsArray>;
+
+type TGetIndicatorHeightRatio = (chart_height: number, indicator_count: number) => TRatio;
+
+type TInitialChartData = {
+    activeSymbols?: ActiveSymbols;
+    masterData?: TQuote[];
+    tradingTimes?: TradingTimes;
+};
+
+type TLanguage = {
+    icon: JSX.Element;
+    key: string;
+    name: string;
+};
+
+type TNotification = {
+    category: string;
+    text: string;
+    type?: 'info' | 'warning' | 'success' | 'error' | 'warning';
+};
+
+type TOHLCData = {
+    close: string;
+    epoch: number;
+    granularity: TicksHistoryRequest['granularity'];
+    high: string;
+    id: string;
+    low: string;
+    open: string;
+    open_time: number;
+    symbol: string;
+};
+
+type TSettings = {
+    activeLanguages?: Array<string | TLanguage> | null;
+    countdown?: boolean;
+    enabledNavigationWidget?: boolean;
+    historical?: boolean;
+    isAutoScale?: boolean;
+    isHighestLowestMarkerEnabled?: boolean;
+    lang?: string;
+    language?: string;
+    minimumLeftBars?: number;
+    position?: string;
+    theme?: string;
+    whitespace?: number;
+};
+
+type TQuote = {
+    Close: number;
+    Date: string;
+    DT?: Date;
+    High?: number;
+    ohlc?: TOHLCData;
+    Open?: number;
+    Low?: number;
+    tick?: TickSpotData;
+    prevClose?: number;
+    Volume?: number;
+};
+
+type TRatio = {
+    height: number;
+    percent: number;
+};
+
+export type TOpenMarket = {
+    category?: string;
+    subcategory?: string | null;
+    market?: string | null;
+};
+
+type TChartTitleProps = {
+    containerId?: string;
+    enabled?: boolean;
+    portalNodeId?: string;
+    searchInputClassName?: string;
+    open?: boolean;
+    open_market?: TOpenMarket | null;
+    isNestedList?: boolean;
+    onChange?: (e: { target: { name: string; value: unknown } }) => Promise<void>;
+};
+
+type TChartModeProps = {
+    portalNodeId?: string;
+    onChartType: (chartType?: string) => void;
+    onGranularity: (granularity?: number) => void;
+};
+
+type TToolbarWidgetProps = React.PropsWithChildren<{ position?: string }>;
+
+type TRefData = {
+    setPosition: ({ epoch, price }: Record<string, number | null | undefined>) => void;
+    div: HTMLDivElement;
+    value?: Element | null;
+};
+
+type TFastMarkerProps = React.PropsWithChildren<{
+    markerRef?: (ref: TRefData | null) => void;
+    threshold?: number;
+    className: string;
+    offsetTop?: number;
+    offsetLeft?: number;
+    x?: number;
+    xPositioner?: string;
+    yPositioner?: string;
+    overlap_y_axis?: boolean;
+}>;
+
+type TPropsWithPortalNodeId = {
+    portalNodeId?: string;
+};
+
+type TSmartChartProps = React.PropsWithChildren<{
```

Comment:
types from the removed SmartChartSwitcher https://github.com/kate-deriv/deriv-app/pull/45/files#diff-04bb2774644da792dc593a913b0892dcd4d21c260415ddda81814bd041f85bb1